CMake/lrs_options.cmake

```diff
@@ -57,3 +57,4 @@ option(ENABLE_SECURITY_FLAGS "Enable additional compiler security flags to enhan
 option(USE_EXTERNAL_LZ4 "Use externally build LZ4 library instead of building and using the in this repo provided version" OFF)
 option(BUILD_ASAN "Enable AddressSanitizer" OFF)
 mark_as_advanced(BUILD_ASAN)
+option(BUILD_WITH_SYSTEM_NLOHMANN_JSON "Use system provided nlohmann_json library" OFF)
```

Comment:
Look, there is `USE_EXTERNAL_LZ4`. For consistency, I would place the addition next to that option, with the name `USE_EXTERNAL_NLOHMANN_JSON`.